Cython/Compiler/ExprNodes.py

```diff
@@ -1973,32 +1973,34 @@ def coerce_to(self, dst_type, env):
     def declare_from_annotation(self, env, as_target=False):
         """Implements PEP 526 annotation typing in a fairly relaxed way.
 
-        Annotations are ignored for global variables, Python class attributes and already declared variables.
+        Annotations are ignored for global variables and already declared variables.
         String literals are allowed and ignored.
         The ambiguous Python types 'int' and 'long' are ignored and the 'cython.int' form must be used instead.
         """
-        if not env.directives['annotation_typing']:
-            return
-        if env.is_module_scope or env.is_py_class_scope:
-            # annotations never create global cdef names and Python classes don't support them anyway
-            return
         name = self.name
-        if self.entry or env.lookup_here(name) is not None:
-            # already declared => ignore annotation
-            return
-
         annotation = self.annotation
-        if annotation.expr.is_string_literal:
-            # name: "description" => not a type, but still a declared variable or attribute
-            atype = None
-        else:
-            _, atype = annotation.analyse_type_annotation(env)
-        if atype is None:
-            atype = unspecified_type if as_target and env.directives['infer_types'] != False else py_object_type
-        if atype.is_fused and env.fused_to_specific:
-            atype = atype.specialize(env.fused_to_specific)
-        self.entry = env.declare_var(name, atype, self.pos, is_cdef=not as_target)
-        self.entry.annotation = annotation.expr
+        entry = self.entry or env.lookup_here(name)
+        if not entry:
+            # annotations never create global cdef names
+            if env.is_module_scope:
```

Comment:
According to [PEP-526](https://www.python.org/dev/peps/pep-0526/#runtime-effects-of-type-annotations), the module scope has an `__annotations__` dict, too. Probably something we can fix up after this PR, though.

Reply:
Should I file a ticket for the work?

Reply:
Sure, so that we don't forget about it if it's not getting done right away.

Reply:
https://github.com/cython/cython/issues/3838